File: evidence/generated-systems/angular/manufacturing-on-angular/src/app/models/InventoryItem.js

// Define collection and schema for InventoryItem
export interface InventoryItem {
    quantityOnHand:
	type : Quantity
    quantityReserved:
	type : Quantity
    lotNumber:
	type : LotId
    serialNumber:
	type : SerialId
    Item:
	type : Schema.Types.ObjectId
    Location:
	type : Schema.Types.ObjectId
#
    collection: 'inventoryItems'
}

File: evidence/generated-systems/django/manufacturing-on-django/manufacturingOnDjango/models/InventoryItem.py
from django.db import models

#======================================================================
# 
# Encapsulates data for model InventoryItem
#
# @author Harbormaster Dev Team
#
#======================================================================

#======================================================================
# Class InventoryItem Declaration
#======================================================================
class InventoryItem (models.Model):

#======================================================================
# attribute declarations
#======================================================================
	quantityOnHand = Quantity
	quantityReserved = Quantity
	lotNumber = LotId
	serialNumber = SerialId
	item = models.ForeignKey('Item', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	location = models.ForeignKey('Location', on_delete=models.CASCADE, null=True, blank=True, related_name='+')

#======================================================================
# function declarations
#======================================================================
	def toString(self):
		str = ""
		str = str + self.quantityOnHand
		str = str + self.quantityReserved
		str = str + self.lotNumber
		str = str + self.serialNumber
		return str;
    
	def __str__(self):
		return self.toString();

	def identity(self):
		return "InventoryItem";
    
	def objectType(self):
		return "InventoryItem";
